packages/item/src/vaadin-item-mixin.js

```diff
@@ -73,96 +76,71 @@ export const ItemMixin = (superClass) =>
       if (attrValue !== null) {
         this.value = attrValue;
       }
-
-      this.addEventListener('focus', () => this._setFocused(true), true);
-      this.addEventListener('blur', () => this._setFocused(false), true);
-      this.addEventListener('mousedown', () => {
-        this._setActive((this._mousedown = true));
-        const mouseUpListener = () => {
-          this._setActive((this._mousedown = false));
-          document.removeEventListener('mouseup', mouseUpListener);
-        };
-        document.addEventListener('mouseup', mouseUpListener);
-      });
-      this.addEventListener('keydown', (e) => this._onKeydown(e));
-      this.addEventListener('keyup', (e) => this._onKeyup(e));
     }
 
-    /** @protected */
-    disconnectedCallback() {
-      super.disconnectedCallback();
-
-      // in Firefox and Safari, blur does not fire on the element when it is removed,
-      // especially between keydown and keyup events, being active at the same time.
-      // reproducible in `<vaadin-select>` when closing overlay on select.
-      if (this.hasAttribute('active')) {
-        this._setFocused(false);
+    /**
+     * Override native `focus` to set focused attribute
+     * when focusing the item programmatically.
+     * @protected
+     * @override
+     */
+    focus() {
+      if (this.disabled) {
+        return;
       }
+
+      super.focus();
+      this._setFocused(true);
+    }
+
+    /**
+     * @param {KeyboardEvent | MouseEvent} _event
+     * @protected
+     * @override
+     */
+    _shouldSetActive(event) {
+      return !this.disabled && !event.defaultPrevented;
```

Comment:
In the new implementation, the `active` attribute won't be set not only in case a keyboard event has been prevented but also a mouse event. I wonder if it might be considered as not expected behavior?

Reply:
Good catch, updated to only take `keydown` events into account.